x/staking/side_handler.go

```diff
@@ -100,11 +99,13 @@ func SideHandleMsgValidatorJoin(ctx sdk.Context, msg types.MsgValidatorJoin, k k
 	signer := pubkey.Address()
 
 	// check signer pubkey in message corresponds
-	if !bytes.Equal(eventLog.SignerPubkey, pubkey.Bytes()[1:]) {
+	expectedPubKey, err := helper.CompressPubKey(eventLog.SignerPubkey)
+
+	if err != nil || !bytes.Equal(expectedPubKey, pubkey.Bytes()) {
 		k.Logger(ctx).Error(
 			"Signer Pubkey does not match",
 			"msgPubKey", pubkey.String(),
-			"compressdPubKey", hmTypes.BytesToHexBytes(eventLog.SignerPubkey),
+			"compressedPubKey", string(expectedPubKey),
```

Comment:
Shouldn't we have hex instead of string(bytes)?

Reply:
string is enough it is just for logging